s3_scrubber/src/lib.rs

```diff
@@ -100,6 +107,57 @@ impl RootTarget {
     }
 }
 
+pub struct BucketConfig {
+    pub region: String,
+    pub bucket: String,
+
+    /// Use SSO if this is set, else rely on AWS_* environment vars
+    pub sso_account_id: Option<String>,
+}
+
+impl Display for BucketConfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}/{}/{}",
+            self.sso_account_id.as_deref().unwrap_or("<none>"),
+            self.region,
+            self.bucket
+        )
+    }
+}
+
+impl BucketConfig {
+    pub fn from_env() -> anyhow::Result<Self> {
+        let sso_account_id = env::var("SSO_ACCOUNT_ID")
+            .context("'SSO_ACCOUNT_ID' param retrieval")
```

Comment:
```suggestion
```

You are adding a context to the error here, to then immediately throw it away (mapping all possible errors to `None`).

Reply:
Fixed.  This was a hangover from when SSO_ACCOUNT_ID was mandatory.